File: examples-sdk/zig/012-basic-plugin-database/main.zig
// 012-basic-plugin-database — Zig SDK equivalent
// Compile: vil compile --from zig --input 012-basic-plugin-database/main.zig --release

const vil = @import("vil");

pub fn main() void {
    var server = vil.Server.init("plugin-database", 8080);
    var plugin_db = vil.Service.init("plugin-db");
    plugin_db.endpoint("GET", "/", "index");
    plugin_db.endpoint("GET", "/plugins", "list_plugins");
    plugin_db.endpoint("GET", "/config", "show_config");
    plugin_db.endpoint("GET", "/products", "list_products");
    plugin_db.endpoint("POST", "/tasks", "create_task");
    plugin_db.endpoint("GET", "/tasks", "list_tasks");
    plugin_db.endpoint("GET", "/pool-stats", "pool_stats");
    plugin_db.endpoint("GET", "/redis-ping", "redis_ping");
    server.service(&plugin_db);
    server.compile();
}

File: examples-sdk/go/012-basic-plugin-database/main.go
// 012-basic-plugin-database — Go SDK equivalent
// Compile: vil compile --from go --input 012-basic-plugin-database/main.go --release
package main

import vil "github.com/OceanOS-id/vil-go"

func main() {
	s := vil.NewServer("plugin-database", 8080)

	plugin_db := vil.NewService("plugin-db")
	plugin_db.Endpoint("GET", "/", "index")
	plugin_db.Endpoint("GET", "/plugins", "list_plugins")
	plugin_db.Endpoint("GET", "/config", "show_config")
	plugin_db.Endpoint("GET", "/products", "list_products")
	plugin_db.Endpoint("POST", "/tasks", "create_task")
	plugin_db.Endpoint("GET", "/tasks", "list_tasks")
	plugin_db.Endpoint("GET", "/pool-stats", "pool_stats")
	plugin_db.Endpoint("GET", "/redis-ping", "redis_ping")
	s.Service(plugin_db)

	s.Compile()
}
